### PR TITLE
Revert session reconnect; remove send via Option

### LINKED ISSUE #16475: [Service Bus] Remove the SendVia option on sender in favor of upcoming implicit transactional via sender and receiver
The current Microsoft.Azure.ServiceBus library has this SendVia option that one can set on the sender in order to use the transactions across entities. As seen in https://github.com/Azure/azure-sdk-for-java/pull/16942, we are moving to an implicit transactional via sender model where the user does not have to provide the SendVia option anymore.

This issue is to remove this option from the newer Azure.Messaging.ServiceBus library so that in a future update, we can add the implicit transactional via sender model without needing to deprecate the SendVia option.

Corresponding PR in Java: https://github.com/Azure/azure-sdk-for-java/pull/17018

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/16475
- leaving SenderOptions as internal for now as we will likely need to add it back soon.

Also reverts the automatic reconnect behavior for session receivers that was added in https://github.com/Azure/azure-sdk-for-net/pull/13735. The reason for this is by reconnecting session receivers we lose some determinism in that users may face session cannot be locked, or service timeout errors when just receiving. The model for sessions is that once you have successfully accepted a session, you should not get those errors, so the change from the referenced PR was counterproductive.

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs [hunk 1]
```diff
@@ -122,15 +115,13 @@ public bool IsClosed
                 Argument.AssertNotNullOrWhiteSpace(entityPath, nameof(entityPath));
                 connection.ThrowIfClosed();
 
-                options = options?.Clone() ?? new ServiceBusSenderOptions();
+                options = options ?? new ServiceBusSenderOptions();
```
Comment (MiYanni) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs:118]: Even though it is removed from the public surface, why would we stop cloning it?

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSenderOptions.cs [hunk 2]
```diff
@@ -9,17 +9,8 @@ namespace Azure.Messaging.ServiceBus
     /// The set of options that can be specified when creating a <see cref="ServiceBusSender"/>
     /// to configure its behavior.
     /// </summary>
-    public class ServiceBusSenderOptions
+    internal class ServiceBusSenderOptions
```
Comment (MiYanni) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSenderOptions.cs:12]: I'm assuming we are just keeping this type for potential future use?

**Review by MiYanni — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs:118]: It has no properties now.. I was going to just leave it, but I was getting an error about it needing to be static. I will just override this error and add back the Clone.

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSenderOptions.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSenderOptions.cs:12]: Yes

**Pull request merged**